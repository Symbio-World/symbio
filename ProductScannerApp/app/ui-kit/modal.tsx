import React from 'react'
import { Modal as RNModal, View, TouchableWithoutFeedback } from 'react-native'
import Icon from 'react-native-vector-icons/MaterialIcons'
import { t, color } from 'react-native-tailwindcss'

type Props = {
  children?: React.ReactNode
  onDismiss?: () => void
}

export const Modal: React.FC<Props> = ({ children, onDismiss }) => (
  <RNModal visible={true} animationType="slide" onRequestClose={onDismiss}>
    <View style={[t.bgGray500, t.p2, t.opacity50, t.absolute, t.z10, t.wFull]}>
      <TouchableWithoutFeedback onPress={onDismiss}>
        <Icon name="arrow-back" size={44} color={color.white} />
      </TouchableWithoutFeedback>
    </View>
    <View style={[t.flex1]}>{children}</View>
  </RNModal>
)
