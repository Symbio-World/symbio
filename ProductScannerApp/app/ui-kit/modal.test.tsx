import 'react-native'
import React from 'react'
import { render, fireEvent } from 'react-native-testing-library'
import { Text, TouchableWithoutFeedback } from 'react-native'
import { Modal } from './modal'

describe('Modal', () => {
  it('renders correctly', () => {
    const { toJSON } = render(
        <Modal>
          <Text>test</Text>
        </Modal>,
      )
    expect(toJSON()).toMatchSnapshot()
  })

  it('triggers onDismiss', () => {
    const handleDismiss = jest.fn()
    const { getByType } = render(<Modal onDismiss={handleDismiss} />)
    fireEvent.press(getByType(TouchableWithoutFeedback))
    expect(handleDismiss.mock.calls.length).toBe(1)
  })
})
