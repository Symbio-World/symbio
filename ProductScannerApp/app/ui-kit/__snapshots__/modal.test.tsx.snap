// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders correctly 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  visible={true}
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#a0aec0",
        },
        Object {
          "padding": 8,
        },
        Object {
          "opacity": 0.5,
        },
        Object {
          "position": "absolute",
        },
        Object {
          "zIndex": 10,
        },
        Object {
          "width": "100%",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 44,
          },
          undefined,
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      î—„
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
      ]
    }
  >
    <Text>
      test
    </Text>
  </View>
</Modal>
`;
